### PR TITLE
Use `-> None` in script templates by default

### PR DESCRIPTION
**Pull request by sobolevn**
### Description

When working together with `mypy` we have a problem with the default template.
`mypy` raises errors on all migrations when used with https://mypy.readthedocs.io/en/stable/config_file.html#confval-disallow_untyped_defs

So, there are several options:
1. Remove this flag for `[mypy-db.alembic.*]` in `mypy.ini`. This is not a very good idea, because custom code that might get written in this directory might be not fully typed.
2. Edit our own template. This is what we end up doing right now, but this requires some manual work and many users might miss it
3. Update the source! I think that this is harmless (because only python3.6+ is supported), but allows better type-checking

### Checklist

Openning this as a draft for now. Will fix tests after I can see what is failing in the CI.
Issue is also on its way! 🙂 

This pull request is:

- [ ] A documentation / typographical error fix
	- Good to go, no issue or tests are needed
- [ ] A short code fix
	- please include the issue number, and create an issue if none exists, which
	  must include a complete example of the issue.  one line code fixes without an
	  issue and demonstration will not be accepted.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.   one line code fixes without tests will not be accepted.
- [x] A new feature implementation
	- please include the issue number, and create an issue if none exists, which must
	  include a complete example of how the feature would look.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.

Fixes https://github.com/sqlalchemy/alembic/issues/764

**Have a nice day!**

### CODE REVIEW

**Review by CaselIT — APPROVED** [review]
looks fine to me

File: alembic/templates/multidb/script.py.mako [hunk 1]
```diff
@@ -19,11 +19,11 @@ branch_labels = ${repr(branch_labels)}
 depends_on = ${repr(depends_on)}
 
 
-def upgrade(engine_name):
+def upgrade(engine_name) -> None:
```
Comment (zzzeek) [inline alembic/templates/multidb/script.py.mako:22]: we should annotate engine_name here also

```py
def upgrade(engine_name: str) -> None:
   ...
```

File: alembic/templates/multidb/script.py.mako [hunk 2]
```diff
@@ -19,11 +19,11 @@ branch_labels = ${repr(branch_labels)}
...
     globals()["upgrade_%s" % engine_name]()
 
 
-def downgrade(engine_name):
+def downgrade(engine_name) -> None:
```
Comment (zzzeek) [inline alembic/templates/multidb/script.py.mako:26]: and here

**Review by zzzeek — CHANGES_REQUESTED** [review]
yes, definitely good idea.   if we can just fully annotate each file, it'll be good. thanks!

**Comment by zzzeek** [comment]
plus we should make sure env.py as given is fully annotated

**Comment by zzzeek** [comment]
and there's one test failing that depends upon the exact syntax of the script, see the build output, that needs to be fixed also.

**Comment by sobolevn** [comment]
I've addressed your comments. Sorry for the long wait! 👍

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **CaselIT** to try to get revision a056551c29acf6c1b452477a2a383de90a1c6e24 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
New Gerrit review created for change a056551c29acf6c1b452477a2a383de90a1c6e24: https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/3798

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **CaselIT** to try to get revision acb04921d96e3921a148397159f238aab50020d4 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
Patchset acb04921d96e3921a148397159f238aab50020d4 added to existing Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/3798

**Comment by sobolevn** [comment]
Done! I hope the format is correct 🙂

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **zzzeek** to try to get revision c1d17b202f414a97e6215cc2a513509462a9db09 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
Patchset c1d17b202f414a97e6215cc2a513509462a9db09 added to existing Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/3798

**Comment by zzzeek** [comment]
very nice thanks

**Comment by sqla-tester** [comment]
**mike bayer** (zzzeek) wrote:

perfect, 10/10

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/3798

**Comment by sqla-tester** [comment]
Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/3798 has been **merged**. Congratulations! :)

**Pull request closed without merging**